src/extensions/replay/sessionrecording.ts

```diff
@@ -1198,6 +1232,39 @@ export class SessionRecording {
         }
     }
 
+    private _pauseRecording() {
+        if (this.status === 'paused') {
+            return
+        }
+        logger.info(LOGGER_PREFIX + ' recording paused due to URL blocker')
+
+        this._tryAddCustomEvent('recording paused', { reason: 'url blocker' })
+
+        this._urlBlocked = true
+
+        // Clear the snapshot timer since we don't want new snapshots while paused
+        clearInterval(this._fullSnapshotTimer)
+
+        // Running this in a timeout to ensure we can
+        setTimeout(() => {
+            this._flushBuffer()
+        }, 100)
+    }
+
+    private _resumeRecording() {
+        if (this.status !== 'paused') {
+            return
+        }
+
+        this._urlBlocked = false
+
+        this._tryTakeFullSnapshot()
+
+        this._scheduleFullSnapshot()
+        this._tryAddCustomEvent('recording resumed', { reason: 'left blocked url' })
```

Comment:
Same deal as the paused event, this one never makes it to `onRRwebEmit` 🤔 
@pauldambra, do you have any idea what I'm doing wrong here?